node/src/database/rocksdb.rs

```diff
@@ -7,130 +7,350 @@
+
+        anyhow::Ok(())
     }
 
-    // fn fetch_block_by_state_root(&self, from_height: u64, state_root: &[u8])
-    // -> Result<&Block> {     // Code for fetching block by state root here
-    //     // ...
-    // }
+    fn fetch_block(&self, hash: &[u8]) -> Result<Option<Block>> {
+        if let Some(blob) = self.snapshot.get_cf(self.ledger_cf, hash)? {
+            let b = Block::read(&mut &blob[..])?;
+            return Ok(Some(b));
+        }
 
-    fn fetch_registry(&self) -> Result<Registry> {
-        anyhow::Ok(Registry::default())
+        // Block not found
+        Ok(None)
     }
 
-    fn fetch_current_height(&self) -> Result<u64> {
-        anyhow::Ok(0)
+    /// Deletes all items from both CF_LEDGER and CF_CANDIDATES column families
+    fn clear_database(&self) -> Result<()> {
+        // Create an iterator over the column family CF_LEDGER
+        let iter = self
+            .inner
+            .iterator_cf(self.ledger_cf, rocksdb_lib::IteratorMode::Start);
+
+        // Iterate through the CF_LEDGER column family and delete all items
+        iter.map(Result::unwrap)
+            .map(|(key, _)| {
+                self.inner.delete_cf(self.ledger_cf, key);
+            })
+            .collect::<Vec<_>>();
+
+        self.clear_candidates()?;
+        anyhow::Ok(())
     }
 
-    fn fetch_block_height_since(
-        &self,
-        since_unix_time: i64,
-        offset: u64,
-    ) -> Result<u64> {
-        anyhow::Ok(0)
+    fn store_candidate_block(&self, b: Block) -> Result<()> {
+        let mut serialized = vec![];
+        b.write(&mut serialized)?;
+
+        self.inner
+            .put_cf(self.candidates_cf, b.header.hash, serialized)?;
+
+        Ok(())
     }
 
-    fn clear_candidate_messages(&mut self) -> Result<()> {
-        anyhow::Ok(())
+    fn fetch_candidate_block(&self, hash: &[u8]) -> Result<Option<Block>> {
+        if let Some(blob) = self.snapshot.get_cf(self.candidates_cf, hash)? {
+            let b = Block::read(&mut &blob[..])?;
+            return Ok(Some(b));
+        }
+
+        // Block not found
+        Ok(None)
     }
 
-    fn clear_database(&mut self) -> Result<()> {
+    /// Deletes all items from CF_CANDIDATES column family
+    fn clear_candidates(&self) -> Result<()> {
+        // Create an iterator over the column family CF_CANDIDATES
+        let iter = self
+            .inner
+            .iterator_cf(self.candidates_cf, rocksdb_lib::IteratorMode::Start);
+
+        // Iterate through the CF_CANDIDATES column family and delete all items
+        iter.map(Result::unwrap)
+            .map(|(key, _)| {
+                self.inner.delete_cf(self.candidates_cf, key);
+            })
+            .collect::<Vec<_>>();
+
         anyhow::Ok(())
     }
 
-    fn commit(&mut self) -> Result<()> {
+    fn commit(self) -> Result<()> {
+        if let Err(e) = self.inner.commit() {
+            return Err(anyhow::Error::new(e).context("failed to commit"));
+        }
```

Comment:
You can collapse this applying the context directly

```suggestion
        self.inner.commit().with_context("failed to commit")?;
```

Reply:
Does not compile

Reply:
Weird... we'll handle it in the future